docs/migrating.md

```diff
@@ -0,0 +1,57 @@
+# Migrating from v1alpha1 to v1alpha2
+
+This document describes changes that were introduced in v1alpha2 API and how to update your templates to the new version.
+
+## ImageMeta -> imageURL conversion
```

Comment:
One more suggestion: we could have headers in the doc, similar to https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/book/src/developer/providers/migrations/v1.4-to-v1.5.md#cluster-api-v14-compared-to-v15 and for example for API changes like below it could be https://github.com/kubernetes-sigs/cluster-api/blob/main/docs/book/src/developer/providers/migrations/v1.4-to-v1.5.md#changes-by-kind